### PR TITLE
In new Go API, name internal metrics based on sender mode.

### PR DESCRIPTION
**Pull request by keep94**

### CODE REVIEW

File: senders/client_factory.go [hunk 1]
```diff
@@ -2,11 +2,12 @@ package senders
 
 import (
 	"fmt"
-	"github.com/wavefronthq/wavefront-sdk-go/internal"
 	"net/url"
 	"os"
 	"strconv"
 	"strings"
+
+	"github.com/wavefronthq/wavefront-sdk-go/internal"
```
Comment (deosu) [inline senders/client_factory.go:10]: Can we revert this change?

File: senders/client_factory.go [hunk 2]
```diff
@@ -44,6 +45,18 @@ type configuration struct {
 	SDKMetricsTags       map[string]string
 }
 
+func (c *configuration) Direct() bool {
```
Comment (deosu) [inline senders/client_factory.go:48]: I am not sure about the naming convention that we follow, but usually, in java, we use something like `isDirectIngested`.

**Review by deosu — APPROVED** [review]
Just a few small comments, but otherwise LGTM.

File: senders/client_factory.go (hunk 1 above)
Comment (keep94) [reply senders/client_factory.go:10]: Generally with go, standard library imports are listed first followed by a space and then 3rd party imports.

File: senders/client_factory.go (hunk 2 above)
Comment (keep94) [reply senders/client_factory.go:48]: Thanks for catching this. Although it is allowed to write methods like GetXXX, it usually isn't done that way in GO.  See https://go.dev/doc/effective_go#Getters.  I am assuming that not having GetXXX methods in Go carries over to not having IsXXX methods in Go as well.
